pkg/build/generator/generator.go

```diff
@@ -211,6 +211,9 @@ func (g *BuildGenerator) checkLastVersion(bc *buildapi.BuildConfig, lastVersion
 // updateImageTriggers sets the LastTriggeredImageID on all the ImageChangeTriggers on the BuildConfig and
 // updates the From reference of the strategy if the strategy uses an ImageStream or ImageStreamTag reference
 func (g *BuildGenerator) updateImageTriggers(ctx kapi.Context, bc *buildapi.BuildConfig, from, triggeredBy *kapi.ObjectReference) error {
+	if triggeredBy == nil {
```

Comment:
Isn't triggeredBy is nil because the ICT has no "From" reference and the BuildStrategy also has no From reference?

Thus, this should have been a validation error when the buildconfig was created.